### PR TITLE
sharp

### PR DESCRIPTION
**Pull request by orthagonal**

### CODE REVIEW

File: lib/optimize.js [hunk 1]
```diff
@@ -1,25 +1,8 @@
...
-  if (options.webp !== false) {
-    plugins.push(imageminWebp());
-  }
+module.exports = async(buffer) => {
   try {
-    return imagemin.buffer(buffer, { plugins });
+    return sharp(buffer).toBuffer();
   } catch (err) {
     // anything other than error 99 is considered an unrecoverable error:
     if (err.code !== 99) {
```
Comment (ecwillis) [inline lib/optimize.js:8]: Does this err code still apply? What does it do? We should make sure this block still works.

File: lib/resize.js [hunk 2]
```diff
@@ -1,19 +1,23 @@
...
-    const color = new TinyColor(background);
-    jimpImage.background(parseInt(color.toHex8(), 16));
+    resizeOptions.background = background;
+  }
+  try {
+    return sharp(buffer)
+      .resize(resizeOptions)
+      .toBuffer();
+  } catch (e) {
+    console.log(e);
```
Comment (ecwillis) [inline lib/resize.js:21]: What goes on after this? Are we just swallowing the error?

File: lib/resize.js [hunk 3]
```diff
@@ -1,19 +1,23 @@
...
   // if only one of width/height was provided, make it a square or let Jimp decide if it is able
-  if (width && !height) {
-    height = type === 'resize' ? Jimp.AUTO : width;
+  const resizeOptions = { fit: type };
+  if (width) {
+    resizeOptions.width = width;
   }
-  if (height && !width) {
-    width = type === 'resize' ? Jimp.AUTO : height;
+  if (height) {
```
Comment (ecwillis) [inline lib/resize.js:10]: similar to the comment above ... what happens in sharp if only one dimension is passed? Need to make sure it's handled.

**Review by ecwillis — CHANGES_REQUESTED** [review]
What about tests? There are test files in here that don't seem updated...

File: lib/optimize.js (hunk 1 above)
Comment (ecwillis) [reply lib/optimize.js:16]: Just so I'm clear ... do we know what happens when sharp throws an error? Does their error object have a code? If all errors are un recoverable, shouldn't we just remove the try / catch block and let all errors throw? It seems to me like we should match this code with the library, and not leave snippets lying around that don't have meaning.

**Review by ecwillis — CHANGES_REQUESTED** [review]

File: lib/resize.js (hunk 3 above)
Comment (orthagonal) [reply lib/resize.js:9]: Sharp automatically resolves the missing dimension.

**Review by ecwillis — APPROVED** [review]

**Pull request merged**